datafusion/sql/src/query.rs

```diff
@@ -62,10 +63,11 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                 // The functions called from `set_expr_to_plan()` need more than 128KB
                 // stack in debug builds as investigated in:
                 // https://github.com/apache/datafusion/pull/13310#discussion_r1836813902
-                let min_stack_size = recursive::get_minimum_stack_size();
```

Comment:
here is pretty good evidence that the feature flag is not working as expected, as this code requires `recursive` to compile but it wasn't failing the build -- I pulled the logic its own structure

Reply:
Late LGTM. `StackGuard` is a very clever fix.